### PR TITLE
feat: add tree command (build, artifacts)

### PR DESCRIPTION
**Pull request by Doozers**

### CODE REVIEW

File: go/pkg/yolotree/yolotree.go [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package yolotree
```
Comment (moul) [inline go/pkg/yolotree/yolotree.go:1]: This package is not very relevant; it adds more complexity

You should implement this in the yolopb package as a method of *yolopb.Batch:

```go
func (b *yolopb.Batch) DisplayTreeFormat() {
```

File: go/pkg/yolotree/yolotree.go [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+import (
+	"berty.tech/yolo/v2/go/pkg/yolopb"
+	"fmt"
+	"sort"
+)
+
+func DisplayTreeFormat(d *yolopb.Batch) {
```
Comment (moul) [inline go/pkg/yolotree/yolotree.go:9]:
```suggestion
func DisplayTreeFormat(b *yolopb.Batch) {
```

File: go/pkg/yolotree/yolotree.go [hunk 3]
```diff
@@ -0,0 +1,22 @@
+package yolotree
+
+import (
+	"berty.tech/yolo/v2/go/pkg/yolopb"
+	"fmt"
+	"sort"
```
Comment (moul) [inline go/pkg/yolotree/yolotree.go:6]:
```suggestion
	"berty.tech/yolo/v2/go/pkg/yolopb"
	"fmt"
	"sort"
```

Your editor won't sort imports, you should config it.

File: go/pkg/yolotree/yolotree.go [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+)
+
+func DisplayTreeFormat(d *yolopb.Batch) {
+	sort.Slice(d.Builds, func(i, j int) bool {
+		return d.Builds[i].FinishedAt.Before(*d.Builds[j].FinishedAt)
+	})
+	for _, b := range d.Builds {
+		fmt.Printf(" - builds: %s from %s\n", b.ShortID, b.GetHasProjectID())
```
Comment (moul) [inline go/pkg/yolotree/yolotree.go:14]: It’s better if the helper returns a string instead of printing.

File: go/cmd/yolo/main.go [hunk 5]
```diff
@@ -276,6 +277,43 @@ func yolo(args []string) error {
...
+
+			svc, err := yolosvc.NewService(db, yolosvc.ServiceOpts{
+				Logger:  logger,
+				DevMode: true,
+			})
+			if err != nil {
+				return err
+			}
+
+			ctx := context.Background()
```
Comment (moul) [inline go/cmd/yolo/main.go:303]: You already have a `ctx` (first arg of the func)

File: go/cmd/yolo/main.go [hunk 6]
```diff
@@ -276,6 +277,43 @@ func yolo(args []string) error {
...
+			input := &yolopb.DevDumpObjects_Request{
+				WithPreloading: withPreloading,
```
Comment (moul) [inline go/cmd/yolo/main.go:305]:
```suggestion
				WithPreloading: true,
```

`tree` without this makes no sense I guess.

File: go/cmd/yolo/main.go [hunk 7]
```diff
@@ -276,6 +277,43 @@ func yolo(args []string) error {
...
+			}
+			ret, err := svc.DevDumpObjects(ctx, input)
+			if err != nil {
+				return err
+			}
+			yolotree.DisplayTreeFormat(ret.Batch)
```
Comment (moul) [inline go/cmd/yolo/main.go:311]:
```suggestion
			fmt.Print(ref.Batch.TreeFormat())
```

File: go/pkg/yolotree/yolotree.go (hunk 3 above)
Comment (Doozers) [reply go/pkg/yolotree/yolotree.go:6]: done :)

**Review by moul — APPROVED** [review]

**Pull request merged**